engine/src/main/java/org/pentaho/di/trans/steps/injector/InjectorMeta.java

```diff
@@ -260,7 +261,7 @@ public void check( List<CheckResultInterface> remarks, TransMeta transMeta, Step
   public void check( List<CheckResultInterface> remarks, TransMeta transMeta, StepMeta stepMeta,
     RowMetaInterface prev, String[] input, String[] output, RowMetaInterface info ) {
     check( remarks, transMeta, stepMeta, prev, input, output, info, transMeta, repository, repository != null
-      ? repository.getMetaStore() : null );
+      ? MetaStoreConst.getDefaultMetastore() : null );
```

Comment:
same, why check repository here?

Reply:
Same as before I just replaced all calls to getRepositoryMetastore with getDefaultMetastore where I could without making extra changes